test/signUpEndpoint.test.js

```diff
@@ -0,0 +1,157 @@
+const request = require('supertest');
+const Chance = require('chance');
+
+const app = require('../app');
+
+const chance = new Chance();
+
+const firstRightQuery = {
+  firstName: chance.first(),
+  lastName: chance.last(),
+  email: chance.email({ domain: 'wolox.com.ar' }),
+  password: chance.string({ length: 10, pool: 'asdfghjkl147258369' })
+};
+
+const secondRightQuery = {
+  firstName: chance.first(),
+  lastName: chance.last(),
+  email: chance.email({ domain: 'wolox.com.ar' }),
+  password: chance.string({ length: 10, pool: 'asdfghjkl147258369' })
+};
+
+const notWoloxEmailQuery = {
+  firstName: chance.first(),
+  lastName: chance.last(),
+  email: chance.email({ domain: 'gmail.com.ar' }),
+  password: chance.string({ length: 10, pool: 'asdfghjkl147258369' })
+};
+
+const notEmailFormatQuery = {
+  firstName: chance.first(),
+  lastName: chance.last(),
+  email: chance.string(),
+  password: chance.string({ length: 10, pool: 'asdfghjkl147258369' })
+};
+
+const notNameQuery = {
+  lastName: chance.last(),
+  email: chance.string(),
+  password: chance.string({ length: 10, pool: 'asdfghjkl147258369' })
+};
+
+const notLastNameQuery = {
+  firstName: chance.first(),
+  email: chance.string(),
+  password: chance.string({ length: 10, pool: 'asdfghjkl147258369' })
+};
+
+const notEmailQuery = {
+  firstName: chance.first(),
+  lastName: chance.last(),
+  password: chance.string({ length: 10, pool: 'asdfghjkl147258369' })
+};
+
+const notPasswordQuery = {
+  firstName: chance.first(),
+  lastName: chance.last(),
+  email: chance.string()
+};
+
+const notValidPasswordQuery = {
+  firstName: chance.first(),
+  lastName: chance.last(),
+  email: chance.string(),
+  password: chance.string({ length: 10, pool: 'as!"7258369' })
+};
+
+const notLongEnoughPasswordQuery = {
+  firstName: chance.first(),
+  lastName: chance.last(),
+  email: chance.string(),
+  password: chance.string({ length: 3, pool: 'as!"7258369' })
+};
+
+test('Right Only One Creation User', async () => {
+  await request(app)
+    .post('/users')
+    .send(firstRightQuery)
+    .expect(201);
+});
+
+test('Try To Create Two User With Equal Email', async () => {
+  await request(app)
+    .post('/users')
+    .send(firstRightQuery)
+    .expect(201);
+  await request(app)
+    .post('/users')
+    .send(firstRightQuery)
+    .expect(400);
```

Comment:
Good, you are checking the number state of error.

But you can check the message also. Why? because the error can be 400 but with a message that is not expected. Add this in fail cases